__tests__/endpoints/getCostumes.test.js

```diff
@@ -0,0 +1,40 @@
+const {
+    getPool,
+    createTables, 
+    createCostume, 
+} = require('../../db/');
+
+const {
+    getBallroomGown
+} = require('../../db/utils');
+
+const {server} = require('../../app')
+const supertest = require('supertest');
+const request = supertest(server)
+
+
+let pool = getPool();
+
+// Double-check that correct database is being used. 
+if (pool.options.database !== 'costume_shop_db_test') {
+    throw new Error("Pool instance was not assigned testing database. Testing aborted. Be sure that NODE_ENV environment variable is set to 'test'.")
+}
```

Comment:
I don't think it's necessary or useful to test this at the top of every test file - for now, I would recommend a separate unit test entirely for testing that you're connected to the right DB. Whether we are connected to the test DB is not relevant to this file. You are essentially testing that pool is setup correctly, which isn't the business of this test.

The reason here is that once your project is deployed, it would have a separate deployment server, with settings set in environment variables. This prevent your tests from connecting to the production server, since no production server pointers would be configured in your code, only through env variables set on the deployment.

Reply:
I'm saying that you will not ever want tests on the prod DB anyway - so the risk of running tests in the wrong DB is very low here. The only risk is whether it accidentally runs on the dev DB. The db url will not come to your app from the environment variables in the deployment app you use, so unless you later manually copy-paste the DB address and then added it to your app, it would basically not be possible for your tests to accidentally write to prod.

Maybe we can actually deploy this API so you can get a feel for how that works :) Always good to get the extra practice

Reply:
I just remembered that the `getPool()` function assigns either the dev or test db based on `process.env.NODE_ENV`. Does it make sense to also test that formally in a Jest test suite?

Reply:
That is what you are testing anyway when you test what is stored on pool? I don't think it's necessary to test this right now to proceed though, I think you can remove the test. Again, you should not be running the test command on the prod server